app/models/incident.rb

```diff
@@ -94,6 +94,20 @@ def claim_for(user)
     save! validate: false
   end
 
+  def export_to_claims!
+    return unless completed?
+    begin
+      ci = ClaimsIncident.create claims_fields table: :incident
+      update claims_id: ci.UID
```

Comment:
Because there are no model level validations, the record either enters the database, or there is an error. This is intentional. That way I don't have to rely on having model validations which are consistent with the constraints of a database that I don't own. So the intended design is to catch any error that occurs within this block and handle it by reporting.

Reply:
That is to say, there's no state where the record doesn't persist but there's no exception. It either enters the database, or there is an error.